quil-rs/src/program/error/mod.rs

```diff
@@ -27,21 +27,16 @@ pub use result::{disallow_leftover, map_parsed, recover};
 pub use syntax::SyntaxError;
 
 /// Errors that may occur while parsing a [`Program`](crate::program::Program).
-// TODO: [`Program`] should have its own error type: https://github.com/rigetti/quil-rs/issues/150
 #[derive(Debug, PartialEq)]
-pub enum ProgramError<T> {
+pub enum ProgramParsingError<T> {
```

Comment:
The smallest nit -- `ParseProgramError` seems like a more conventional name to me. I think it's because it's `(AttemptedAction)Error` but that's not something I've thought about before.

Reply:
I looked it up out of curiosity, and [this Rust API Guidelines page](https://rust-lang.github.io/api-guidelines/naming.html#names-use-a-consistent-word-order-c-word-order) seems to be defining the convention I mentioned.

Reply:
Yeah, I like that better. I renamed it.